airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py

```diff
@@ -237,11 +237,12 @@ def read_records(
 
 
 class Profiles(IncrementalKlaviyoStream):
-    """Docs: https://developers.klaviyo.com/en/reference/get_profiles"""
+    """Docs: https://developers.klaviyo.com/en/v2023-02-22/reference/get_profiles"""
 
     cursor_field = "updated"
     api_revision = "2023-02-22"
     page_size = 100
+    state_checkpoint_interval = 100  # API can return maximum 100 records per page
```

Comment:
I'm a little bit afraid of this change as we don't use sorting in the API endpoints. How can we validate that the records are emitted in the order of the cursor field? If they are not, this is dangerous as we could have:
* emit record with cursor value `10`
* emit a state message with cursor value `10`
* emit record with cursor value `5`
* the sync crash

In that case, we wouldn't lose the record with cursor value `5` but we can see that if the crash happens just before, we would have lost the record.

Reply:
I believe this is only applied to the streams which have sorting by cursor field at the API level - we add sorting param here: https://github.com/airbytehq/airbyte/pull/32291/files#diff-eb5f2206f720a1d96317e5a8b62e21de2eaa12b401666efccb798362b337eaf3R148

Reply:
Given all the checkpoint_interval you added are for IncrementalKlaviyoStream, everything looks good then. 

One non blocking question though: why the comment next to the checkpoint interval? How is the page size linked to the checkpoint_interval?

Reply:
The connector is emitting its state after every page it's read. If the page is not full, the state will be emitted at the end of the read, otherwise, since pages are sorted by cursor, we can checkpointing after each page.

Reply:
This logic has been changed somewhat recently : we are emitting a state message after X number of records not after X number of records in a slice. Does that change the way you look at this problem?

Reply:
No, since there are no overridden `stream_slices`